satpy/modifiers/parallax.py

```diff
@@ -313,3 +320,26 @@ def __call__(self, projectables, optional_datasets=None, **info):
         local_scene = global_scene.resample(plax_corr_area)
         local_scene["dataset"].attrs["area"] = base_area
         return local_scene["dataset"]
+
+
+def _get_satpos_alt(cth_dataset):
+    """Get satellite position if no orbital parameters in metadata.
+
+    Some cloud top height datasets lack orbital parameter information in
+    metadata.  Here, orbital parameters are calculated based on the platform
+    name and start time, via Two Line Element (TLE) information.
+
+    Needs pyorbital, skyfield, and astropy to be installed.
+    """
+    from pyorbital.orbital import tlefile
+    from skyfield.api import EarthSatellite, load
+    from skyfield.toposlib import wgs84
+
+    tle = tlefile.read(name := cth_dataset.attrs["platform_name"])
+    es = EarthSatellite(tle.line1, tle.line2, name)
+    ts = load.timescale()
+    gc = es.at(ts.from_datetime(
+        cth_dataset.attrs["start_time"].replace(tzinfo=datetime.timezone.utc)))
+    (lat, lon) = wgs84.latlon_of(gc)
+    height = wgs84.height_of(gc).to("km")
+    return (lon.degrees, lat.degrees, height.value)
```

Comment:
Should this be moved to a different module or even a different package, such as pyorbital?